pay-api/src/pay_api/services/eft_short_name_historical.py

```diff
@@ -0,0 +1,194 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service to manage EFT Short names historical data."""
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+from decimal import Decimal
+from typing import Optional
+
+from sqlalchemy import and_, case, exists, false, func, select
+from sqlalchemy.orm import aliased
+
+from pay_api.models import EFTShortnamesHistorical as EFTShortnamesHistoricalModel
+from pay_api.models import PaymentAccount as PaymentAccountModel
+from pay_api.models import db
+from pay_api.models.eft_short_names_historical import EFTShortnameHistorySchema
+from pay_api.utils.enums import EFTHistoricalTypes
+from pay_api.utils.user_context import user_context
+from pay_api.utils.util import unstructure_schema_items
+
+
+@dataclass
+class EFTShortnameHistory:  # pylint: disable=too-many-instance-attributes
+    """Used for creating EFT Short name historical data."""
+
+    short_name_id: int
+    amount: Decimal
+    credit_balance: Decimal
+    payment_account_id: Optional[int] = None
+    related_group_link_id: Optional[int] = None
+    statement_number: Optional[int] = None
+    hidden: Optional[bool] = False
+    is_processing: Optional[bool] = False
+
+
+@dataclass
+class EFTShortnameHistorySearch:
+    """Used for searching EFT Short name historical records."""
+
+    page: Optional[int] = 1
+    limit: Optional[int] = 10
+
+
+class EFTShortnameHistorical:
+    """Service to manage EFT Short name historical data."""
+
+    @staticmethod
+    def create_funds_received(history: EFTShortnameHistory) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name funds received historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by='SYSTEM',
+            credit_balance=history.credit_balance,
+            description='Funds Received',
```

Comment:
What's the point of description if your transaction_type is the same thing?

Reply:
Type for processing, description for display purposes. Descriptions have been tweaked multiple times on UX, so I separated it and allow for additional information to be added if needed.

Reply:
Should be done at the UI level not the backend

Reply:
Will drop the column

Reply:
column dropped, updated UI to reflect this